wagtail/admin/widgets/button_select.py

```diff
@@ -2,17 +2,17 @@
 from django.utils.translation import gettext_lazy as _
 
 
-class ButtonSelect(forms.Select):
+class ButtonSelect(forms.RadioSelect):
```

Comment:
Just thinking - do we even need this widget at all?

Maybe the `ButtonSelect` can be completely removed as this is now behaving essentially the exact same as RadioSelect

Reply:
OK, sure I'll remove the extra function... can you let me know how to rectify the failed CI tests?

Reply:
It's the formatting that's wrong. You'll need to run the formatter and push a new commit. 

You can either run the commands manually
https://docs.wagtail.org/en/latest/contributing/python_guidelines.html#pep8

Or set up your pre commit hook
https://docs.wagtail.org/en/latest/contributing/developing.html#automatically-lint-and-code-format-on-commits

I know the development guidelines are long but most stuff should be there in that contributing section

Reply:
I wanted to state that when I'm removing the  ButtonSelect feature errors are showing up in 

`wagtail\wagtail\contrib\redirects\filters.py` 

and then to resolve that I'll have to be adding similar kind of functionality as ButtonSelect in the filters.py. So is it essential to remove the ButtonSelect function?

Reply:
If the ButtonSelect is basically the exact same html and behaviour as the RadioSelect built in widget. Then delete ButtonSelect and update all usage to be the RadioSelect.

Reply:
ok sure

Reply:
I'm trying to commit the new changes but there's being some issue with the pre-commit...
```
git commit -m "Replaced ButtonSelect with RadioSelect"
[INFO] Installing environment for https://github.com/pre-commit/mirrors-prettier.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:\\Users\\hp\\AppData\\Local\\Programs\\Python\\Python310\\python.exe', '-mnodeenv', '--prebuilt', '--clean-src', '\\\\?\\C:\\Users\\hp\\.cache\\pre-commit\\repoufa88ptd\\node_env-system', '-n', 'system')
return code: 1
stdout: (none)
stderr:
    Installing system node.js on win32 is not supported!

Check the log at C:\Users\hp\.cache\pre-commit\pre-commit.log
```

And I have node installed in my system and my system specs are 64 bit as well... I don't know why this is coming... I tried re writing the ```.pre-commit-config``` file as well ... 

Can I get a little assistance in this matter.

Reply:
I was setting up my pre-commit and came across this error 

```ModuleNotFoundError: No module named 'resource'```

And after a bit of googling I found out that resource package is only for unix not for Windows .... then how should I proceed further?

Reply:
For the pre-commit issue - I recommend raising a new issue with your problems, it's very likely that the pre-commit hook is not really set up to consider Windows devices.

It's best to also remove those changes from your PR.

You can use `make format` to format the Python code but may need to get `make` working on Windows.
See https://docs.wagtail.org/en/stable/contributing/developing.html#development-on-windows

Reply:
I tried using the make command but it didn't work on windows.

I raised the issue for pre-commit #10172

Reply:
Thanks for raising, were you able to run `make-format` though?

It should be able to be run irrespective of the pre-commit hook.